src/Cms/Environment.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Filesystem\F;
+use Kirby\Http\Server;
+use Kirby\Http\Uri;
+
+/**
+ * The environment object takes care of
+ * secure host and base URL detection, as
+ * well as loading the dedicated
+ * environment options.
+ *
+ * @package   Kirby Cms
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Environment
+{
+    /**
+     * @var string
+     */
+    protected $root;
+
+    /**
+     * @var \Kirby\Http\Server
+     */
+    protected $server;
+
+    /**
+     * @var \Kirby\Http\Uri
+     */
+    protected $uri;
+
+    /**
+     * @param \Kirby\Http\Server $server
+     * @param string $root
+     * @param bool|string|array|null $allowed
+     */
+    public function __construct(Server $server, string $root, $allowed = null)
+    {
+        $this->root   = $root;
+        $this->server = $server;
```

Comment:
Why do we need to pass a server object - the server class is static methods only (it's already a bit weird on the App class).

Reply:
You are actually right. We could just use the static methods instead. I wonder if we will ever work with a real instance, but then we need to refactor quite a bit of code anyway. I will simplify it.